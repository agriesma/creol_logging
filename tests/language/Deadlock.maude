load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Deadlock" : Class | Inh: noInh, Param: noVid, Att: noSubst,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "l" |-> null, "_" |-> null,
              Code: call( "l" ; "this" ; "n" ; emp ) ; release ; skip ;
              get( "l" ; noVid ) ; free( "l" ) ; return ( emp ) >,
            < "n" : Method | Param: noVid, Att: "_" |-> null, Code: release ;
              return ( emp ) >,
      Ocnt: 0 > } .
endm

