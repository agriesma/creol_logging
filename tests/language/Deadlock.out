==========================================
reduce in PROGRAM : state .
result State: {
< "Deadlock" : Class | Inh: noInh, Param: noVid, Att: noSubst, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "n" : Method | Param: noVid, Att: "_" |-> null, Code: release ;
    return(emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null, "l" |->
    null, Code: call("l" ; "this" ; "n" ; emp) ; release ; skip ; get("l" ;
    noVid) ; free("l") ; return(emp) >,
  Ocnt: 0 >
}
Bye.
